src/cobra/io/web/biomodels_repository.py

```diff
@@ -0,0 +1,105 @@
+"""Provide functions for loading metabolic models over the wire."""
+
+
+import gzip
+from io import BytesIO
+from typing import List, Union
+
+import httpx
+import pydantic
+
+from .abstract_model_repository import AbstractModelRepository
+
+
+class BioModelsFile(pydantic.BaseModel):
+    """Define a single BioModels file description."""
+
+    name: str
+    size: int = pydantic.Field(alias="fileSize")
+
+
+class BioModelsFilesResponse(pydantic.BaseModel):
+    """Define the BioModels files JSON response."""
+
+    main: List[BioModelsFile] = []
+
+
+class BioModels(AbstractModelRepository):
+    """
+    Define a concrete implementation of the BioModels repository.
+
+    Attributes
+    ----------
+    name : str
+        The name of the BioModels repository.
+
+    """
+
+    name: str = "BioModels"
+
+    def __init__(
+        self,
+        **kwargs,
+    ) -> None:
+        """
+        Initialize a BioModels repository interface.
+
+        Other Parameters
+        ----------------
+        kwargs
+            Passed to the parent constructor in order to enable multiple inheritance.
+
+        """
+        super().__init__(url="https://www.ebi.ac.uk/biomodels/model/", **kwargs)
+
+    def get_sbml(self, model_id: str) -> bytes:
+        """
+        Attempt to download an SBML document from the repository.
+
+        Parameters
+        ----------
+        model_id : str
+            The identifier of the desired metabolic model. This is typically repository
+            specific.
+
+        Returns
+        -------
+        bytes
+            A gzip-compressed, UTF-8 encoded SBML document.
+
+        Raises
+        ------
+        httpx.HTTPError
+            In case there are any connection problems.
+
+        """
+        data = BytesIO()
+        response = httpx.get(
+            url=self._url.join(f"files/{model_id}"),
+            headers={"Accept": "application/json"},
+        )
+        response.raise_for_status()
+        files = BioModelsFilesResponse.parse_obj(response.json())
+        is_found = False
+        for model in files.main:
+            if model.name.endswith("xml"):
+                is_found = True
+                break
+        if not is_found:
+            RuntimeError(f"'{model_id}' does not seem to contain an SBML document.")
```

Comment:
Python has a `for` `else` for this use case which is much cleaner to read, also `model_id` -> `model` in RunTimeError
``` 
        is_found = False
        for model in files.main:
            if model.name.endswith("xml"):
                is_found = True
                break
        if not is_found:
            RuntimeError(f"'{model_id}' does not seem to contain an SBML document.")
``` 
=>
```
        for model in files.main:
            if model.name.endswith("xml"):
                break
        else:
            RuntimeError(f"'{model}' does not seem to contain an SBML document.")
```

Reply:
In addition you have to check that the file is an SBML file. There are other XML files in the Combine archives on biomodels such as SED-ML files which also have xml endings.

Reply:
I didn't know about `for`... `else` that's awesome. Thanks for teaching me something new about Python :slightly_smiling_face: 

> In addition you have to check that the file is an SBML file. There are other XML files in the Combine archives on biomodels such as SED-ML files which also have xml endings.

My observation about the BioModels response was that under the `"main"` key, you only have the model file, all the other files are in the `"additional"` key (which I ignore). https://www.ebi.ac.uk/biomodels/docs/#/operations/Model-related%20operations/get_model_files__modelId_ Do you know this to be false?

Reply:
Did not know about the `main` flag. The flag should clearly identify the SBML model, so no need to check if the `*.xml` is an SBML in this case.